File: DSA in C/FirstNonRepeating.c
# include<stdio.h>
int main(){
    int n;
    printf("Enter number of elements for array:\n");
    scanf("%d", &n);
    int arr[n];
    for(int i=0;i<n;i++){
        scanf("%d", &arr[i]);
    }
    int freq[1000000] = {0};

    for(int i=0;i<n;i++){
        freq[arr[i]]++;
    }
    int found = -1;

    for(int i=0;i<n;i++){
        if(freq[arr[i]] ==1){
            found = arr[i];
            break;
        }
    }
    
    if(found ==-1){
        printf("No non repeating element found.\n");
    }else{
        printf("First non repeating element is: %d", found);
    }
    return 0;
}

File: Python and OOPS/FirstNonRepeating.py
def first_non_repeating(s):
    s = s.replace(' ', '').lower()
    freq = {}
    for i in s:
        freq[i] = freq.get(i,0) +1
    for ch in s:
        if freq[ch] == 1:
            return ch
    return None
s = input('Enter your string:')
result = first_non_repeating(s)
if result :
    print(f'First non repeating character in you string is: {result}')
else:
    print('No first non repeating character in your stiring.')